cli/command_restore.go

```diff
@@ -41,32 +47,125 @@ directory ID and optionally a sub-directory path. For example,
 var (
 	restoreCommand           = app.Command("restore", restoreCommandHelp)
 	restoreCommandSourcePath = restoreCommand.Arg("source-path", restoreCommandSourcePathHelp).Required().String()
-	restoreCommandTargetPath = restoreCommand.Arg("target-path", "Path of the directory for the contents to be restored").Required().String()
 
+	restoreTargetPath           = ""
 	restoreOverwriteDirectories = true
 	restoreOverwriteFiles       = true
+	restoreMode                 = restoreModeAuto
+)
+
+const (
+	restoreModeLocal         = "local"
+	restoreModeAuto          = "auto"
+	restoreModeZip           = "zip"
+	restoreModeZipNoCompress = "zip-nocompress"
+	restoreModeTar           = "tar"
+	restoreModeTgz           = "tgz"
 )
 
 func addRestoreFlags(cmd *kingpin.CmdClause) {
+	cmd.Arg("target-path", "Path of the directory for the contents to be restored").Required().StringVar(&restoreTargetPath)
 	cmd.Flag("overwrite-directories", "Overwrite existing directories").BoolVar(&restoreOverwriteDirectories)
-	cmd.Flag("overwrite-files", "Specifies whether or not to overwrite already existing files").
-		BoolVar(&restoreOverwriteFiles)
+	cmd.Flag("overwrite-files", "Specifies whether or not to overwrite already existing files").BoolVar(&restoreOverwriteFiles)
+	cmd.Flag("mode", "Override restore mode").EnumVar(&restoreMode, restoreModeAuto, restoreModeLocal, restoreModeZip, restoreModeZipNoCompress, restoreModeZip, restoreModeTar)
```

Comment:
`restoreModeZip` is listed twice here. `restoreModeTgz` isn't listed

Reply:
May be worth adding test cases where the mode is explicitly provided by the flag

Reply:
This one is simple to fix
```suggestion
	cmd.Flag("mode", "Override restore mode").EnumVar(&restoreMode, restoreModeAuto, restoreModeLocal, restoreModeZip, restoreModeZipNoCompress, restoreModeTar, restoreModeTgz)
```

Reply:
thanks!

Reply:
also added more test cases